lib/fluent/plugin/out_s3.rb

```diff
@@ -107,7 +107,8 @@ def write(chunk)
           "path" => path,
           "time_slice" => chunk.key,
           "file_extension" => @compressor.ext,
-          "index" => i
+          "index" => i,
+          "uuid_flush" => uuid_random
```

Comment:
From benchmark, calling uuid_random at each flush doesn't impact in fluentd performance.

### result

```
% ruby uuid_test.rb
       user     system      total        real
   0.290000   0.000000   0.290000 (  0.293195)
   0.070000   0.000000   0.070000 (  0.064821)
   0.310000   0.000000   0.310000 (  0.310315)
```

### code

```rb
require 'benchmark'
require 'uuidtools'

num = 10000
s3_object_key_format = "%{path}%{time_slice}_%{index}.%{file_extension}"
s3_object_key_format_with_uuid = "%{path}%{time_slice}_%{index}.%{file_extension}.%{uuid_flush}"

Benchmark.bm do |x|
  x.report {
    num.times {
      values_for_s3_object_key = {
        "path".freeze => "logs/".freeze,
        "time_slice".freeze => "1234".freeze,
        "file_extension".freeze => 'txt'.freeze,
        "index".freeze => 0,
        "uuid_flush".freeze => UUIDTools::UUID.random_create.to_s
      }
      s3_object_key_format.gsub(%r(%{[^}]+})) { |expr|
        values_for_s3_object_key[expr[2...expr.size-1]]
      }
    }
  }
  x.report {
    num.times {
      values_for_s3_object_key = {
        "path".freeze => "logs/".freeze,
        "time_slice".freeze => "1234".freeze,
        "file_extension".freeze => 'txt'.freeze,
        "index".freeze => 0
      }
      s3_object_key_format.gsub(%r(%{[^}]+})) { |expr|
        if expr == "%{uuid_flush}".freeze
          UUIDTools::UUID.random_create.to_s
        else
          values_for_s3_object_key[expr[2...expr.size-1]]
        end
      }
    }
  }
  x.report {
    num.times {
      values_for_s3_object_key = {
        "path".freeze => "logs/".freeze,
        "time_slice".freeze => "1234".freeze,
        "file_extension".freeze => 'txt'.freeze,
        "index".freeze => 0
      }
      s3_object_key_format_with_uuid.gsub(%r(%{[^}]+})) { |expr|
        if expr == "%{uuid_flush}".freeze
          UUIDTools::UUID.random_create.to_s
        else
          values_for_s3_object_key[expr[2...expr.size-1]]
        end
      }
    }
  }    
end
```